mantidproject/mantid
# Make WorkspaceProperty::isDefault() to work with workspaces not in the ADS

This PR fixes a case where, when one sets a `WorkspaceProperty` to a workspace which is not in the ADS, `WorkspaceProperty::isDefault()` would still return `true`. This enables checking if the user has given a non-ADS workspace for an optional input property by `isDefault`.

**To test:**

Create a mock algorithm using the first script provided in the issue. The second script should raise an error in Mantid logs before these changes are applied while after applying, the mock algorithm should run fine. 

Fixes #20790.

**Release Notes** 

The user facing changes to the Python API are mentioned in the Framework's release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
